### ISSUE TITLE
Generate functions to use for refetchQueries

### ISSUE THREAD

**Issue by brandon-leapyear** [dotansimha/graphql-code-generator#3478]
**Is your feature request related to a problem? Please describe.**

The [`refetchQueries`](https://www.apollographql.com/docs/react/data/mutations/#options) seems super useful for us in contrast with update functions, since we would mostly be duplicating server logic. However, we are not fond of the fact that the variables aren't type checked to match the query:

```ts
{
  refetchQueries: [
    { query: GetUserDocument, variables: { name: 'user1' } }, // should type check
    { query: GetUserDocument, variables: { foo: 1 } }, // should not type check
  ],
}
```

**Describe the solution you'd like**

It would be great if there could be an option that generates something of the form
```ts
export function refetchGetUserQuery(variables: GetUserQueryVariables) {
  return { query: GetUserDocument, variables: variables }
}
```

**Describe alternatives you've considered**

**Additional context**

**Comment by brandon-leapyear** [comment]
Fixed by #3479

**Comment by ardatan** [comment]
Let's keep it open until new release.

**Comment by dotansimha** [comment]
Available in `v1.13.0`
